packages/rules/src/mustSatisfyPeerDependencies.ts

```diff
@@ -0,0 +1,272 @@
+/*!
+ * Copyright 2020 Palantir Technologies, Inc.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ *
+ */
+
+import { Context, RuleModule } from "@monorepolint/core";
+import { mutateJson, PackageJson } from "@monorepolint/utils";
+import path from "path";
+import * as r from "runtypes";
+import { coerce } from "semver";
+import { IPackageDependencyGraphNode, PackageDependencyGraphService } from "./util/packageDependencyGraphService";
+
+const Options = r.Union(
+  r.Partial({
+    skipUnparseableRanges: r.Undefined,
+  }),
+  r.Record({
+    skipUnparseableRanges: r.Boolean,
+  })
+);
+
+type Options = r.Static<typeof Options>;
+
+export const mustSatisfyPeerDependencies: RuleModule<typeof Options> = {
+  check: checkSatisfyPeerDependencies,
+  optionsRuntype: Options,
+};
+
+/*
+  separating on `|`, this regex allows any of the following formats:
+  - *
+  - x
+  More info: https://docs.npmjs.com/about-semantic-versioning
+*/
+export const MATCH_ANY_VERSION_RANGE = /^(\*|x)$/;
+/*
+  separating on `|`, this regex allows any of the following formats:
+  - 15
+  - ^15
+  - 15.x
+  - ^15.x
+  - 15.x.x
+  - ^15.x.x
+  - ^15.2
+  - ^15.2.x
+  - ^15.2.1
+  More info: https://docs.npmjs.com/about-semantic-versioning
+*/
+export const MATCH_MAJOR_VERSION_RANGE = /^(\^?\d+|\^?\d+\.x|\^?\d+\.x\.x|\^\d+\.\d+|\^\d+\.\d+\.x|\^\d+\.\d+\.\d+)$/;
+// does not accept `<` or `>`
+// TODO: accept minor pins `~4.2.1`
+export const RANGE_REGEX = /^(\*|x|\^?\d+(\.x|\.x\.x|\.\d+|\.\d+\.x|\.\d+\.\d+)?( \|\| \^?\d+(\.x|\.x\.x|\.\d+|\.\d+\.x|\.\d+\.\d+)?)*)$/;
+
+function checkSatisfyPeerDependencies(context: Context, opts: Options) {
+  const { skipUnparseableRanges } = opts;
+  const graphService = new PackageDependencyGraphService();
+  const rootNode = graphService.buildDependencyGraph(path.resolve(context.getPackageJsonPath()), 1);
+  const rootDependencies = rootNode.packageJson.dependencies || {};
```

Comment:
nit: calling all these "root" confused me a bit, because i always viewed these rules as operating on a per-package basis

Reply:
yeah, I was kinda following the format from `PackageDependencyGraphService`. I'd be happy to rename these to `packageDependencies` etc

Reply:
renamed to `package` (and `test` in tests), though `package` may be a bit ambiguous. I can also rename to `current` if you prefer

Reply:
i thought `package` was a reserved word, `pkg` or `currentPackage` sgtm

Reply:
nvm, i see the new name 👍